modules/ballerina-core/src/main/java/org/ballerinalang/connector/impl/ServerConnectorRegistry.java

```diff
@@ -45,12 +47,21 @@
  */
 public class ServerConnectorRegistry {
 
-    private static ServerConnectorRegistry instance = new ServerConnectorRegistry();
+    private static final Map<ProgramFile, ServerConnectorRegistry> CONNECTOR_REGISTRY_MAP = new ConcurrentHashMap<>();
     private Map<String, BallerinaServerConnector> serverConnectorMap = new HashMap<>();
     private boolean initialized = false;
 
-    public static ServerConnectorRegistry getInstance() {
-        return instance;
+    public static ServerConnectorRegistry getInstance(ProgramFile programFile) {
```

Comment:
Why do we need per programfile registry?

Reply:
As per the offline discussion with @lankavitharana we decided to remove this method and access server connector registry via programFile